### ISSUE TITLE
RuntimeWarning with pandas.dataframes that contain np.nan values

### ISSUE THREAD

**Issue by dacoex** [spyder-ide/spyder#2991]
* A large dataframe with missing values is loaded.
* ipython console issues a `RuntimeWarning: ` and is blocked thereafter for any further interactive use

The warning:

```
(C:\WinPython-64bit-3.4.3.7Qt5\python-3.4.3.amd64\lib\site-packages\pandas\core\format.py:2087: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e8).any()
C:\WinPython-64bit-3.4.3.7Qt5\python-3.4.3.amd64\lib\site-packages\pandas\core\format.py:2088: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) &
C:\WinPython-64bit-3.4.3.7Qt5\python-3.4.3.amd64\lib\site-packages\pandas\core\format.py:2089: RuntimeWarning: invalid value encountered in greater
  (abs_vals > 0)).any()

```

System info:

Spyder 3.0.0dev
pd.__version__ -->  '0.17.1'

See also:
* [Pandas error - invalid value encountered](http://stackoverflow.com/questions/30519487/pandas-error-invalid-value-encountered)
* [pandas give a warning when working with np.NaN values](https://github.com/pydata/pandas/issues/9950)

**Comment by ccordoba12** [comment]
> A large dataframe with missing values is loaded.

Sorry, but this is not enough for us to solve this problem. Please give us a simple and reproducible example that we can test in our side to see what's happening.

**Comment by dacoex** [comment]
sure. The question will be to find the part that triggers the warning.
I will load in the standalone console check if it works there.

**Comment by dacoex** [comment]
In the standalone all works fine.

Do you have a suggeston on how to prepare reproducible example?

**Comment by goanpeca** [comment]
It would be very helpfull If you could provide a link to the actual data being loaded in the data frame, plus the exact steps to reproduce it.

**Comment by ccordoba12** [comment]
Yes, that's the idea. Please give the Python code you're trying to run that's generating the freeze you're seeing.

**Comment by dacoex** [comment]
OK, please give me some time to prepare the example. Thanks for your work.

**Comment by rlabbe** [comment]
d = numpy.random.rand(2500000, 50)
    d[0,0] = None
    df  = pd.DataFrame(d)
    df.sort()

Sometimes it locks up, sometimes I can keep going.

I can pretty consistently get it to lock up by typing (with the data above)

    df.head(2) 

Note there is no carriage return at the close parens - as soon as I type the ')' I get the error (pasted below).

df.head(2)c:\Anaconda3\lib\site-packages\pandas\core\format.py:2087: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e8).any()
c:\Anaconda3\lib\site-packages\pandas\core\format.py:2088: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) &
c:\Anaconda3\lib\site-packages\pandas\core\format.py:2089: RuntimeWarning: invalid value encountered in greater
  (abs_vals > 0)).any()

Sometimes I can keep typing after this, sometimes it is locked up.

Following advice in other PRs I have executed

    $ conda install -f pandas

(some say an inconsistant conda install can cause this with version 0.15). 

My environment:

Python 3.5.1 |Anaconda 2.5.0 (64-bit)| (default, Jan 29 2016, 15:01:46) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 4.0.3 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
%guiref   -> A brief reference about the graphical user interface.

import pandas as pd

pd.__version__
Out[2]: '0.17.1'

Spyder: 2.3.8
Windows 10 64 bit

**Comment by ccordoba12** [comment]
@rlabbe, thanks for the code. Does it fail only for big DataFrames, or the problem has to do with the presence of nan's.

**Comment by rlabbe** [comment]
Ah, good point. Just made a 25x50 array. Sort runs with no problem, but the closing parens in head() triggers the same warning. Hmm, interesting, just did it again and this time typing the opening parens caused the warning, and the interface locked up

**Comment by kitjosh1050** [comment]
I was encountering a similar issue when putting pandas dataframes in dictionaries. Now, this only started to happen after I upgraded to latest version of Anaconda/python.

```
3.5.1 |Anaconda 2.5.0 (64-bit)| (default, Jan 29 2016, 15:01:46) [MSC v.1900 64 bit (AMD64)] 
 pandas version =  0.17.1
```
I'm on Windows 7 64bit btw.

Now, I didn't encounter this issue when just working with the dataframe. 

```python
# Case 1 - Similar as rlabbe, very big df, no issues
import pandas as pd
import numpy
c1_d = numpy.random.rand(2500000, 50)
c1_d[0,0] = None
c1_df  = pd.DataFrame(c1_d)
c1_df.head(10)
```

However, I did run into the issue with putting the dataframe into a dictionary (regardless of row count). You have to type the code into the IPython Console. Paste the whole code and all is fine!

Also, after I encountered the issue. Any time I started typing the first parenthesis of **any** pandas method, e.g. `pd.read_csv(` I got the message and had to restart spyder. 

```python
# Case 2 - Num rows doesn't matter, but putting df into dictionary does
c2_d = numpy.random.rand(10, 50)
c2_d[0,0] = None
c2_df  = pd.DataFrame(c2_d)
c2_di = {}
c2_di["c2_df"] = c2_df
c2_di["c2_df"].headC:\Users\Josh.Josh-PC\Anaconda3\lib\site-packages\pandas\core\format.py:2087: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e8).any()
C:\Users\Josh.Josh-PC\Anaconda3\lib\site-packages\pandas\core\format.py:2088: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) &
C:\Users\Josh.Josh-PC\Anaconda3\lib\site-packages\pandas\core\format.py:2089: RuntimeWarning: invalid value encountered in greater
  (abs_vals > 0)).any()
```



I checked to confirm that when there are no nulls, no issue!
```python
# Case 3 - Tried without any null cells and it's cool
c3_d = numpy.random.rand(10, 50)
c3_df  = pd.DataFrame(c3_d)
c3_di = {}
c3_di["c3_df"] = c3_df
c3_di["c3_df"].head(5)
```

**Comment by daniel0720** [comment]
I was encountering the same problem with joshmorel. I updated to latest version of Anaconda/pandas

Python 3.5.1 |Anaconda 2.4.1 (64-bit)| (default, Dec  7 2015, 15:00:12) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 4.0.1 -- An enhanced Interactive Python.

> pd.__version__
>  '0.17.1'

I'm on Windows 7 64bit btw.

My codes are list below.

`import numpy as np`
`import pandas as pd`
`from pandas import Series, DataFrame`
`s=Series(np.random.randn(6))`
`s[::2]=np.nan`

Then I input s.fill<Tab>, problem come.
`C:\Users\daniel\Anaconda3\lib\site-packages\pandas\core\format.py:2087: RuntimeWarning: invalid value encountered in greater  has_large_values = (abs_vals > 1e8).any()`
`C:\Users\daniel\Anaconda3\lib\site-packages\pandas\core\format.py:2088: RuntimeWarning: invalid value encountered in less  has_small_values = ((abs_vals < 10 ** (-self.digits)) &`
`C:\Users\daniel\Anaconda3\lib\site-packages\pandas\core\format.py:2089: RuntimeWarning: invalid value encountered in greater  (abs_vals > 0)).any()`

but  I can input all the code.
even when i started typing the first parenthesis of any pandas method, eg. `s.fillna(`, i got the same problem, and have to restart spyder.

**Comment by ehremington** [comment]
This still happens anytime you press [tab] to use autocomplete

**Comment by ccordoba12** [comment]
We tried to fix this in Spyder 2.3.9. Are you sure you're using that version?

**Comment by ehremington** [comment]
That is what Help > About Spyder tells me. Although conda list shows spyder at version 2.3.9 and spyder-app at 2.3.8 and there is no update available for spyder-app

**Comment by ccordoba12** [comment]
@ehremington, could you post a code example that's giving you the error?

**Comment by ehremington** [comment]
`import numpy as np`
`import pandas as pd`
`s = pd.Series([1,2,np.nan,5,6])`
`s`
Then the comes the error on the next line. If you just begin a statement and hit [tab] to do the autocomplete it give the error in the attached picture. For example, I do:
`df = pd.DataF` and then hit [tab] to complete "DataFrame." When you do it moves that line down and replaces it with:

`/home/bert/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py:2191:`
` RuntimeWarning: invalid value encountered in greater`
 ` has_large_values = (abs_vals > 1e6).any()`
`/home/bert/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py:2192: RuntimeWarning:` `invalid value encountered in less`
`  has_small_values = ((abs_vals < 10**(-self.digits)) &`
`/home/bert/anaconda3/lib/python3.5/site-packages/pandas/formats/format.py:2193: RuntimeWarning:` `invalid value encountered in greater`
 ` (abs_vals > 0)).any()`

![justbeforetabispressed](https://cloud.githubusercontent.com/assets/10223282/15301080/92852436-1b71-11e6-95ba-5f524092a80a.png)

![rightaftertabispressed](https://cloud.githubusercontent.com/assets/10223282/15301087/991316a0-1b71-11e6-893b-bd1d77d26dad.png)

**Comment by ccordoba12** [comment]
Thanks for the detailed report @ehremington!! Yours seems to be a slightly different problem than the one I fixed here (I targeted the exact error of the original reporter :-)

Could you tell me what's your Pandas version?

**Comment by ehremington** [comment]
0.18.1

It also happens in IPython console when I open parenthesis

**Comment by MarcosCarreira** [comment]
I have found the same problem:
- Loading large dataframes
- These probably have missing values
<img width="641" alt="screen shot 2016-06-16 at 16 05 48" src="https://cloud.githubusercontent.com/assets/14150204/16129738/639d0048-33dc-11e6-9f52-9a5d8dc584b0.png">
The DataFrame variables do not appear in the Variable Explorer, and sometimes the prompt will not accept a Return, treating it as new line.
Setup: OSX 10.11.5, Anaconda Navigator 1.1.1 Spyder 2.3.9

**Comment by MarcosCarreira** [comment]
Another example (prompt):
<img width="456" alt="screen shot 2016-06-16 at 16 15 02" src="https://cloud.githubusercontent.com/assets/14150204/16129986/8e2033d4-33dd-11e6-96ab-ce2e613044b1.png">

**Comment by jitseniesen** [comment]
@MarcosCarreira as I understand it, your issue was addressed in commit 0676fac which was added to the development version about two weeks ago. The fix should be included in the next beta release of Spyder.

**Comment by MarcosCarreira** [comment]
@jitseniesen That's great!
Thanks
Marcos

**Comment by MarcosCarreira** [comment]
Running spyder 3.0.0b4 with some large DataFrames (some with NaN), still found same issue.

**Comment by CellCS** [comment]
i met the same problem about  NaN, but i just update spyder and conda update anaconda. 
![image](https://cloud.githubusercontent.com/assets/16886624/19624293/2a48d8bc-98bd-11e6-82e8-2c1fa4b567e5.png)

**Comment by jitseniesen** [comment]
@CellCS Your picture does not look like Spyder :) It looks like this is a warning raised by pandas when you call `describe()` on a DataFrame with NaN, which has nothing to do with Spyder.

**Comment by ling60** [comment]
I encountered the same problem
![img](https://cloud.githubusercontent.com/assets/9345401/19636918/b637d0e8-99fe-11e6-87fd-fe6bd4eaffa1.jpg)

**Comment by jitseniesen** [comment]
@ling60 Try upgrading Spyder. The fix is only included in versions 2.3.9 and upward, including the new and shiny versions 3.0.0 and 3.0.1 :)

**Comment by ling60** [comment]
Thank you

Ling Liu
Associate Professor
School of Economics Information Engineering
Southwestern University of Finance and Economics

刘凌
副教授
西南财经大学经济信息工程学院

**Comment by osherry** [comment]
My version is Spyder 2.3.9.  I also encountered this problem. 
After upgrading to the latest version by running "pip install --upgrade spyder" in cmd window, the version remains to be 2.3.9 and the problem still exists.
The warning:
a=Series([np.nan,2.5],index=['a','b'])
E:\Anaconda3\lib\site-packages\pandas\formats\format.py:2191: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
E:\Anaconda3\lib\site-packages\pandas\formats\format.py:2192: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10**(-self.digits)) &
E:\Anaconda3\lib\site-packages\pandas\formats\format.py:2193: RuntimeWarning: invalid value encountered in greater
  (abs_vals > 0)).any()

**Comment by jitseniesen** [comment]
@osherry @ling60 I was wrong. The fix included in Spyder 2.3.9 is incomplete. You need to upgrade to 3.0.0 or 3.0.1. 不好意思。

> After upgrading to the latest version by running "pip install --upgrade spyder" in cmd window, the version remains to be 2.3.9

We need to fix this to make progress. The issue may be that you initially installed Spyder using Anaconda and then upgraded it using pip. When you subsequently run Spyder again, the computer runs the Anaconda version and not the pip version.

The command "conda list spyder" (in a cmd window) shows the versions installed. If you get something like

```
spyder                    2.3.9                    py35_1  
spyder                    3.0.1                     <pip>
```

then you have spyder 2.3.9 via Anaconda and spyder 3.0.1 via pip. In that case, you need to first uninstall the pip version so that you only have the Anaconda version; the command for that is "pip uninstall spyder". Then upgrade the Anaconda version with the command "conda upgrade spyder". Hopefully that will really update Spyder.
